### PR TITLE
Switch to ci matrix

### PR DESCRIPTION
**Pull request by mnecas**
Continuation of https://github.com/oVirt/ovirt-ansible-collection/pull/447/

### CODE REVIEW

**Review by michalskrivanek — APPROVED** [review]

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -7,68 +7,39 @@ on:
     branches:
       - master
   workflow_dispatch:
+    branches:
+      - master
```
Comment (mwperina) [inline .github/workflows/build.yaml:11]: worklfow_dispatch could without branch definition

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -7,68 +7,39 @@ on:
...
-      - name: Install ansible packages
-        run: yum install -y ansible-core ansible-test glibc-langpack-en
-      - name: Update all packages
-        run: yum update -y
-
-      - name: Install pip modules
-        run: pip3 install pycodestyle pylint==2.4.4 voluptuous yamllint
-
       - name: Checkout
         uses: actions/checkout@v2
```
Comment (mwperina) [inline .github/workflows/build.yaml:30]: I'd rather move checkout after all package installation

File: automation/build.sh [hunk 3]
```diff
@@ -21,9 +21,6 @@ rpmbuild \
     -D "_topmdir $ROOT_PATH/rpmbuild" \
     -ts ./*.gz
 
-# install any build requirements
-yum-builddep $ROOT_PATH/output/*src.rpm
-
```
Comment (mwperina) [inline automation/build.sh:26]: I'd leave it here, without it you cannot add new dependencies to the project and have successfull build

File: .github/workflows/build.yaml (hunk 1 above)
Comment (michalskrivanek) [reply .github/workflows/build.yaml:11]: indeed, no branch here please

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -7,68 +7,39 @@ on:
...
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-
-      - name: Install packages
-        run: yum install -y createrepo_c rpm-build yum-utils git ansible-core ansible-test python3-pip
-      - name: Update all packages
-        run: yum update -y
+      - name: Update pip
+        run: pip3 install pip --upgrade
```
Comment (michalskrivanek) [inline .github/workflows/build.yaml:33]: this should be remove too, buildcontainer has up to date pip from python3-pip

File: .github/workflows/build.yaml (hunk 2 above)
Comment (mnecas) [reply .github/workflows/build.yaml]: ack

File: .github/workflows/build.yaml (hunk 4 above)
Comment (mnecas) [reply .github/workflows/build.yaml:28]: When I was running the build manually it was still needed, looks like the python3-pip is not the latest pip.

Comment (michalskrivanek) [reply .github/workflows/build.yaml:28]: that might be possible, distros usually do not package pipy latest versions...the main question is if you need the latest for whatever reason...

Comment (mnecas) [reply .github/workflows/build.yaml:28]: We still need it when we try to install the `ansible-lint<5.0.0` it throws `No module named 'setuptools_rust'`

File: automation/build.sh (hunk 3 above)
Comment (mwperina) [reply automation/build.sh:26]: No, buildcontainer can provide new dependencies only after the PR adding them is merged and buildcontainer rebuilt. Using above will allow you to test the new dependency change even before it's merged

File: ovirt-ansible-collection.spec.in [hunk 5]
```diff
@@ -11,6 +11,12 @@ License:	ASL 2.0 and GPLv3+
 BuildArch:	noarch
 Url:		http://www.ovirt.org
 
+BuildRequires:	ansible-core
+BuildRequires:	ansible-test
+%if 0%{?rhel} > 7 && 0%{?rhel} < 9
+BuildRequires:	glibc-langpack-en
```
Comment (mwperina) [inline ovirt-ansible-collection.spec.in:17]: What do we need this for?

Comment (mnecas) [reply ovirt-ansible-collection.spec.in:17]: The ansible-test fails with `/bin/sh: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)` without it

Comment (mwperina) [reply ovirt-ansible-collection.spec.in:17]: OK

**Review by mwperina — APPROVED** [review]
OK, we can add `yum-builddep` before we really need to add a new dependency

File: automation/build.sh (hunk 3 above)
Comment (michalskrivanek) [reply automation/build.sh:26]: that chicken and egg problem was the reason for https://github.com/oVirt/buildcontainer/pull/14 :)
no need to merge any temporary code anywhere. gets overwritten nightly.

File: .github/workflows/build.yaml (hunk 4 above)
Comment (michalskrivanek) [reply .github/workflows/build.yaml:28]: yeah, it's quite an old one in there...
# rpm -q python3-pip
python3-pip-9.0.3-22.el8.noarch
# pip3 install pip --upgrade
Successfully installed pip-21.3.1

**Pull request merged**